datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -104,13 +109,48 @@ enum FileStreamState {
     Limit,
 }
 
+struct FileStreamMetrics {
+    /// Time elapsed for file opening
+    pub time_opening: (metrics::Time, Option<Instant>),
+    /// Time elapsed for file scanning + first record batch of decompression + decoding
+    pub time_scanning: (metrics::Time, Option<Instant>),
+    /// Time elapsed for data decompression + decoding
+    pub time_processing: (metrics::Time, Option<Instant>),
+}
+
+impl FileStreamMetrics {
+    fn new(metrics: &ExecutionPlanMetricsSet, partition: usize) -> Self {
+        let time_opening = (
+            MetricBuilder::new(metrics).subset_time("time_elapsed_opening", partition),
+            None,
+        );
+
+        let time_scanning = (
+            MetricBuilder::new(metrics)
+                .subset_time("time_elapsed_scanning_processing", partition),
```

Comment:
I recommend keeping this name consistent with the field (`FileStreamMetrics::time_scanning`) -- if the `processing` is important here, then perhaps we can rename the field

```suggestion
                .subset_time("time_elapsed_scanning", partition),
```